docs/flakty-tests.md

```diff
@@ -0,0 +1,260 @@
+interactions run fully within the same tick of the loop. This eliminates a lot
+of problems which plague Capybara and similar frameworks based on Selenium and
+similar tools. In Capybara we go to great lengths to avoid problems with the
+dreaded `StaleReferenceError`, but we simply do not need to do this in BigTest.
+By running within the event loop, and ensuring that we never release it, we can
+ensure that our elements are never stale.
+
+This has some implications on the way that interactors are defined, but we
+have worked very hard to make it as intuitive as possible to write
+interactors in such a way that they will be very resistant to problems on
+this front. Of course, BigTest will also ship with a rich library of
+interactors already built in, for all of the most common interface elements
+in web applications, so most users never have to define their own interactors
+if they don't want to.
+
+The TextField interactor we used above could be defined like this:
+
+``` typescript
+const TextField = createInteractor<HTMLInputElement>('text field')({
+  selector: 'input',
+  locators: {
+    byId: (element) => element.id,
+  },
+  actions: {
+    fillIn: (element, value: string) => { element.value = value }
+  }
+});
+```
+
+Both the locator and action functions are synchronous functions, which means
+that they do not release the event loop. This allows us to transform them
+into convergent functions on the interactor which perform the entire
+interaction of finding the element and runnign the functions against them.
+
+BigTest is written in TypeScript and our interactors are fully typesafe when
+used from TypeScript both in their definition and when using them.
+
+## Inherent ambiguities
+
+We have claimed that our convergence strategy is enough to eliminate
+flakiness, but as we mentioned as well, having a solid strategy is only part
+of the solution, the other half of the solution is writing the tests in such
+a way that they make maximal usage of this solution and avoid inherent
+ambiguities that we cannot otherwise resolve.
+
+Let's illustrate what we mean by this with an example: Imagine we have a paginated
+list with a 'Next' link at the bottom. Whenever we press the next link we will load
+a new set of rows, we will also update the `href` attribute of the next link to point
+to the next page.
+
+[GRAPHIC]
+
+Now imagine a written a test written like this:
+
+``` typescript
+await Link('Next').click();
+await Link('Next').click();
+```
+
+Which page will we end up on? Intuitively we would say that we should end up
+on the *third* page, but unfortunately the actual answer is "it depends". The
+first click on `Next` will start a request to the server which will complete
+after some time, this will change the `href` of the link. But if our second
+click happens *before* the request completes, we will navigate to the second
+page *twice*, and we actually end up on the *second* page.
+
+[GRAPHIC]
+
+### Guessing
```

Comment:
To provide like a thread to the three subsections I suggest we append some descriptor to each approach. Preferably we'd use descriptors that go from worse to best. 

Something like:
- A naïve approach: Guessing
- A better approach: Anchoring
- The best approach: Fixing the UI

Reply:
I love it!